akka-http-core/src/test/scala/akka/http/impl/engine/client/HttpsProxyGraphStageSpec.scala

```diff
@@ -171,5 +201,27 @@ class HttpsProxyGraphStageSpec extends AkkaSpec {
         fn(source, flowInProbe, flowOutProbe, sink)
       }
     }
+
+    def testCaseWithBasicProxyAuth(fn: (PublisherProbe, SubscriberProbe, PublisherProbe, SubscriberProbe) ⇒ Unit): Unit = {
```

Comment:
This looks almost identical as `testCase`. Couldn't you just add another parameter to `testCase` to carry the authentication?

```scala
def testCase(proxyAuth: Option[HttpCredentials] = None)(fn: ...)
```